src/test/java/htsjdk/samtools/CRAMFileReaderTest.java

```diff
@@ -146,18 +177,27 @@ public void testCRAMReader6_ReferenceRequired() throws IOException {
     public void testCRAMReader6_InputRequired() throws IOException {
         InputStream bis = null;
         File file = null;
-        ReferenceSource refSource = null;
         new CRAMFileReader(bis, file, createReferenceSource(), ValidationStringency.STRICT);
     }
 
     // constructor 7: CRAMFileReader(final File cramFile, final File indexFile, final ReferenceSource referenceSource,
     //                final ValidationStringency validationStringency)
     @Test(description = "Test CRAMReader 7 reference required", expectedExceptions = IllegalArgumentException.class)
     public void testCRAMReader7_ReferenceRequired() throws IOException {
-        InputStream bis = new ByteArrayInputStream(new byte[0]);
-        File file = new File(TEST_DATA_DIR, "cram_with_crai_index.cram");
         ReferenceSource refSource = null;
-        new CRAMFileReader(file, file, refSource, ValidationStringency.STRICT);
+        new CRAMFileReader(CRAM_FILE, CRAM_FILE, refSource, ValidationStringency.STRICT);
     }
 
+    @Test
+    public void testCRAMReader7_ShouldAutomaticallyFindCRAMIndex()throws IOException {
+        File indexFile = null;
+        CRAMFileReader reader = new CRAMFileReader(CRAM_FILE, indexFile, REFERENCE, ValidationStringency.STRICT);
+        Assert.assertTrue(reader.hasIndex(), "Can't find existing CRAM index.");
+    }
+
+    @Test
+    public void testCRAMReader7_ShouldUseCRAMIndex() throws IOException {
+        CRAMFileReader reader = new CRAMFileReader(CRAM_FILE, INDEX_FILE, REFERENCE, ValidationStringency.STRICT);
+        Assert.assertTrue(reader.hasIndex(), "Can't find existing CRAM index.");
+    }
```

Comment:
It would be good to add a negative test for each constructor that got updated just to make sure there are no problems in those code paths. There are plenty of cram files in the test folder with no index.